tidy3d/components/monitor.py

```diff
@@ -667,6 +667,69 @@ def storage_size(self, num_cells: int, tmesh: ArrayLike[float, 1]) -> int:
         return BYTES_COMPLEX * len(self.ux) * len(self.uy) * len(self.freqs) * 4
 
 
+class DiffractionEfficiencyMonitor(FieldMonitor, PlanarMonitor):
+    """:class:`Monitor` that uses a 2D Fourier transform to compute the
+    diffraction efficiency for given (or all) diffraction orders.
+
+    Example
+    -------
+    >>> monitor = DiffractionEfficiencyMonitor(
+    ...     center=(1,2,3),
+    ...     size=(5,5,0),
+    ...     freqs=[250e12, 300e12],
+    ...     name='diffraction_efficiency_monitor',
+    ...     normal_dir='+',
+    ...     orders_x=[0, 1, 2],
+    ...     orders_y=[0],
+    ...     )
+    """
+
+    normal_dir: Direction = pydantic.Field(
+        "+",
+        title="Normal vector orientation",
+        description="Direction of the surface monitor's normal vector w.r.t. "
+        "the positive x, y or z unit vectors. Must be one of ``'+'`` or ``'-'``. "
+        "Defaults to ``'+'`` if not provided.",
+    )
+
+    orders_x: Tuple[int, ...] = pydantic.Field(
+        None,
+        title="Diffraction orders along x",
+        description="Diffraction orders along x for which efficiency should be computed. "
+        "If ``None``, all orders between -1000 and 1000 will be considered.",
```

Comment:
What is the reason behind 1000?

Reply:
up to you. is it likely people will want 100s of diffraction orders? we could also make the default just `[0]` or something.

Reply:
That would be simpler, yeah. I guess if someone does want to put in a large number of orders out of laziness, they can.